### PR TITLE
[New] Add inferno-compat@3.6.3 w/ npm auto-update

### PR DESCRIPTION
**Pull request by maruilian11**
@cdnjs/library-reviewer Please help me check this PR, thanks!
**There are more versions on github, but I only find `dist` on npm package.**

Pull request for issue: #10840

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/infernojs/inferno
 * Official website (optional, not the repository): https://infernojs.org/
 * NPM package url (optional): https://www.npmjs.com/package/inferno-compat
 * License and its reference: MIT, https://github.com/infernojs/inferno/blob/master/LICENSE.md
 * GitHub / Bitbucket popularity (required):
   Watch 304
   Star 10,445
   Fork 443
 * NPM download stats (optional):
   307 downloads in the last day
   5,562 downloads in the last week
   21,049 downloads in the last month
# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by maruilian11** [comment]
@extend1994 Have found built files before v1.2.2 on GitHub repo or the way to build the files? If the built files don't exist, we can't manual build them.

**Comment by extend1994** [comment]
@maruilian11 I meant the versions on npm. Now I know they are not packaged by the author so we can't add them. Sorry for that!

**Comment by maruilian11** [comment]
@PeterDaveHello I just rebased my branch with the latest master branch, please help me review this PR, thanks.

File: ajax/libs/inferno-compat/package.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/infernojs/inferno.git"
+  },
+  "npmName": "inferno-compat",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "inferno-compat*"
```
Comment (PeterDaveHello) [inline ajax/libs/inferno-compat/package.json]: Shouldn't this start with `**/` prefix as it's under `dist`?

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer @sufuf3 I just modified the filemap from `inferno-compat*` to `**/inferno-compat*`, could you please help me review this PR? thanks.

**Comment by sufuf3** [comment]
@maruilian11 maybe we need to update `.gitignore` to ignore `v4.0.0-alpha1*`
I just found this commit: https://github.com/cdnjs/cdnjs/commit/58b487c90bf0d65932d4842b8ccb10bbd763a782

What do you think?
Thank you.

**Comment by maruilian11** [comment]
>Note:
v4.0.0-alpha1-alpha.d24e2d51 can't fit the FileMap, but I think this version is alpha, maybe we can still use current Filemap.

@sufuf3 I didn't see a problem about the filemap. Can you explain more details?

**Comment by sufuf3** [comment]
@maruilian11 sorry about that. Please ignore about `FileMap` part.

But I think maybe we need to update `.gitignore` to ignore `v4.0.0-alpha1*`
Because the library release two newest versions, and I think we don't need to include.
I used `npm view inferno-compat versions`.

**Comment by maruilian11** [comment]
@PeterDaveHello Do we now still need to use `.gitignore` to ignore alpha and beta version?

**Comment by PeterDaveHello** [comment]
Ignoring those dummy versions would be good.

**Comment by sufuf3** [comment]
Because `4.0.0-alpha1-alpha.d24e2d51` can't fit the `mapfile` rule, I thought that maybe we need to use `.gitignore` to ignore the version.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer @sufuf3 I just updated `.gitignore` to ignore the unstable versions. Please help me review this PR, thanks!

**Comment by maruilian11** [comment]
@PeterDaveHello I think we should host the old versions which are already on cdnjs but belong to `inferno`  of cdnjs because users won't know the old versions are under `inferno` if they have found `inferno-compat`. How do you think?

**Comment by PeterDaveHello** [comment]
Let's host them all.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer @sufuf3 Could you help me review this PR again? Thanks.

**Comment by sufuf3** [comment]
As https://github.com/cdnjs/cdnjs/pull/11484#issuecomment-317246589 I think this PR need add v1.0.1 ~ v1.2.2 of inferno-compat.
Maybe need to add another commit to manually add v1.0.1 ~ v1.2.2 of inferno-compat.
What do you think?

Thank you.

**Comment by maruilian11** [comment]
@sufuf3 Thanks. Added them already.

**Comment by extend1994** [comment]
@dakshshah96 Please help review this PR, thank you.

File: .gitignore [hunk 2]
```diff
@@ -204,6 +204,9 @@ node_modules/
 /ajax/libs/slim.js/1.*/
 /ajax/libs/slim.js/2.[0-5].*/
 /ajax/libs/europa/4.0.0-alpha*/
+/ajax/libs/inferno-compat/*-alpha*/
+/ajax/libs/inferno-compat/*-beta*/
+/ajax/libs/inferno-compat/*-rc*/
```
Comment (PeterDaveHello) [inline .gitignore]: Do we need to exclude them all?

Comment (maruilian11) [inline .gitignore]: Do you mean just exlcude `alpha` and `beta` version?

Comment (PeterDaveHello) [inline .gitignore]: Don't we just want to exclude the dummy versions especially those version string contains checksum?

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer I updated the `.gitignore` and add `beta`, `alpha` and `rc` versions back. And add v1.0.1 and v1.0.2 minified files. Please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
@extend1994 Thank you for that. I add the minified file.

**Comment by PeterDaveHello** [comment]
v1.0.0-alpha6/9 missing?

**Comment by PeterDaveHello** [comment]
Shouldn't `inferno-compat.node.js` in v1.3 series be excluded or ignored?

**Pull request merged**